### PR TITLE
Parallel Browser Tests (Firefox & Chrome)

### PR DESCRIPTION
**Pull request by ioedeveloper**

### CODE REVIEW

File: test-browser/commands/executeScript.js [hunk 1]
```diff
@@ -2,15 +2,23 @@ const EventEmitter = require('events')
...
+
     this.api
-      .clearValue('span[data-id="terminalCliInput"]')
-      .click('div[data-id="terminalCli"]')
-      .keys(script)
+      .clearValue('*[data-id="terminalCliInput"]')
+      .click('*[data-id="terminalCli"]')
+    if (runtimeBrowser === 'firefox') {
+      this.api.sendKeys('*[data-id="terminalCliInput"]', script)
+      .sendKeys('*[data-id="terminalCliInput"]', this.api.Keys.ENTER)
```
Comment (yann300) [inline test-browser/commands/executeScript.js]: could you add a specific custom command for that?

File: test-browser/tests/gist.js [hunk 2]
```diff
@@ -35,8 +37,8 @@ module.exports = {
         browser
           .modalFooterCancelClick()
           .executeScript(`remix.loadgist('${gistid}')`)
-          .switchFile('browser/gists')
-          .switchFile(`browser/gists/${gistid}`)
+        if (runtimeBrowser === 'chrome') browser.switchFile('browser/gists')
```
Comment (yann300) [inline test-browser/tests/gist.js]: that's quite unexpected that not running `browser.switchFile('browser/gists')` for firefox still work...

File: test-browser/tests/remixd.js [hunk 3]
```diff
@@ -52,15 +52,20 @@ module.exports = {
...
       .addFile('test_import_node_modules.sol', sources[3]['browser/test_import_node_modules.sol'])
       .clickLaunchIcon('solidity')
       .testContracts('test_import_node_modules.sol', sources[3]['browser/test_import_node_modules.sol'], ['SafeMath'])
+      : ''
   },
   'Import from node_modules and reference a github import': function (browser) {
-    browser
+    const browserName = browser.options.desiredCapabilities.browserName
+
+    return browserName === 'chrome' ? browser
```
Comment (yann300) [inline test-browser/tests/remixd.js]: you don't run this test for firefox?

File: test-browser/tests/terminal.js [hunk 4]
```diff
@@ -26,7 +26,7 @@ module.exports = {
     browser
     .waitForElementVisible('*[data-id="terminalCli"]')
     .click('*[data-id="terminalCli"]')
-    .keys('remix.')
+    .sendKeys('*[data-id="terminalCliInput"]', 'remix.')
```
Comment (yann300) [inline test-browser/tests/terminal.js]: `sendKeys` works for chrome too?
so maybe we don't need https://github.com/ethereum/remix-ide/pull/2626/files#diff-bce217bd304a22784cfcdae7bd97b11aR10

File: test-browser/tests/remixd.js (hunk 3 above)
Comment (ioedeveloper) [reply test-browser/tests/remixd.js]: Yes, the test is dependent on the previous tests you disabled for firefox. It also fails for chrome too when the beginning tests are disabled.

File: test-browser/tests/terminal.js (hunk 4 above)
Comment (ioedeveloper) [reply test-browser/tests/terminal.js]: Yes, i'll take it out and make them uniform.

File: test-browser/tests/gist.js (hunk 2 above)
Comment (ioedeveloper) [reply test-browser/tests/gist.js]: Yeah, I think it has to do with the mutiple entry of `.sendKeys('*[data-id="terminalCliInput"]', this.api.Keys.ENTER')` in `.executeScript` and how firefox handles it.

File: test-browser/tests/remixd.js [hunk 5]
```diff
@@ -52,15 +52,20 @@ module.exports = {
       when a relative import is used (i.e import "openzeppelin-solidity/contracts/math/SafeMath.sol")
       remix (as well as truffle) try to resolve it against the node_modules and installed_contracts folder.
     */
-    browser
+    const browserName = browser.options.desiredCapabilities.browserName
+
+    return browserName === 'chrome' ? browser
```
Comment (yann300) [inline test-browser/tests/remixd.js]: do you have any idea why it fails for firefox?

Comment (ioedeveloper) [reply test-browser/tests/remixd.js]: I get this error `Unable to import "openzeppelin-solidity/contracts/math/SafeMath.sol": File not found` when the first test is disabled for firefox and chrome also.

Comment (yann300) [reply test-browser/tests/remixd.js]: I don't get why the first test was disabled. was it introduced by parallel testing?

Comment (ioedeveloper) [reply test-browser/tests/remixd.js]: No, i think it was introduced by you with these error messages.
```javascript
  var browserName = browser.options.desiredCapabilities.browserName
  if (browserName === 'safari' || browserName === 'internet explorer') {
    console.log('do not run remixd test for ' + browserName + ': sauce labs doesn\'t seems to handle websocket')
    browser.end()
    return
  }
  if (browserName === 'firefox') {
    console.log('do not run remixd test for ' + browserName + ': TODO to reenable later')
    browser.end()
    return
  }
```

Comment (ioedeveloper) [reply test-browser/tests/remixd.js]: I have enabled remixd test for firefox and both tests now passes.

**Pull request merged**